mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -51,8 +51,9 @@ mlir::Attribute addElemToArrayAttr(mlir::PatternRewriter &rewriter,
   return rewriter.getArrayAttr(values);
 }
 
-LogicalResult add(mlir::PatternRewriter &rewriter, mlir::PDLResultList &results,
-                  llvm::ArrayRef<mlir::PDLValue> args) {
+LogicalResult addOrSub(mlir::PatternRewriter &rewriter,
+                       mlir::PDLResultList &results,
+                       llvm::ArrayRef<mlir::PDLValue> args, const bool useAdd) {
```

Comment:
Can you make the `useAdd` a template parameter and use if `constexpr` for checking if it's add/sub?